core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java

```diff
@@ -461,18 +461,20 @@ private boolean add(Block block, boolean tryConnecting,
                 checkState(tryConnecting, "bug in tryConnectingOrphans");
                 log.warn("Block does not connect: {} prev {}", block.getHashAsString(), block.getPrevBlockHash());
                 orphanBlocks.put(block.getHash(), new OrphanBlock(block, filteredTxHashList, filteredTxn));
+                if (tryConnecting) {
```

Comment:
If the checkState() three lines above is "correct", this if() shouldn't be needed. But of course it doesn't hurt either.